test/EndToEndTests/Services/ODataSimplifiedService/ODataSimplifiedDataSource.cs

```diff
@@ -168,17 +169,18 @@ protected override IEdmModel CreateModel()
             return ODataSimplifiedInMemoryModel.CreateModel("Microsoft.OData.SampleService.Models.ODataSimplified");
         }
 
-        protected override void ConfigureContainer(IContainerBuilder builder)
+        protected override void ConfigureContainer(IServiceCollection services)

```

Comment:
I want to point out that, if I'm reading this right, this whole process is a DI anti-pattern. We should not be registering multiple different types of the same settings in the DI container and relying on FIFO or LIFO to resolve what we want.

The registration process should allow people to change the settings however they want via an Action lambda with overloads to the `AddDefaultODataServices` method. That would let people set the different options the right way at the right time without multiple registrations or cloning.

If done right, those settings would even be able to be specified in appsettings.json files and not just in code, because they would be applied by the configuration system.